server/lib/csp.js

```diff
@@ -2,30 +2,42 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-// middleware to take care of CSP. CSP headers are only sent in development
-// mode, and are not sent if the user is running the unit tests.
+// Middleware to take care of CSP. CSP headers are not sent unless config
+// option 'csp.enabled' is set (default true in development), with a special
+// exception for the /tests/index.html path, which are the frontend unit
+// tests.
 
 'use strict';
 
 var helmet = require('helmet');
 var config = require('./configuration');
 
+var SELF = "'self'";
+var DATA = 'data:';
+var GRAVATAR = 'https://secure.gravatar.com';
+
 var cspMiddleware = helmet.csp({
-                      'default-src': ['\'self\''],
-                      'connect-src': [
-                        '\'self\'',
-                        config.get('fxaccount_url'),
-                        config.get('oauth_url'),
-                        config.get('profile_url')
-                      ],
-                      'img-src': [
-                        '\'self\'',
-                        'data:',
-                        config.get('profile_images_url'),
-                        'https://secure.gravatar.com'
-                      ],
-                      'report-uri': '/_/csp-violation'
-                    });
+  'default-src': [
+    SELF
+  ],
+
+  'connect-src': [
+    SELF,
+    config.get('fxaccount_url'),
+    config.get('oauth_url'),
+    config.get('profile_url')
+  ],
+
+  'img-src': [
+    SELF,
+    DATA,
+    GRAVATAR,
+    config.get('profile_images_url')
+  ],
+
+  'report-uri': config.get('csp.reportUri'),
+  reportOnly: config.get('csp.reportOnly')
```

Comment:
I believe the helmet docs say:

> _"You can specify keys in a camel-cased fashion (`imgSrc`) or dashed (`img-src`); they are equivalent."_

I vote for switching all to camelCase since it feels more readable.